### PR TITLE
Automatically set the transaction name.

### PR DESCRIPTION
**Pull request by petermattis**
Added client.Tx.SetDebugName() for additional manual control.

### CODE REVIEW

File: client/db.go [hunk 1]
```diff
@@ -374,8 +376,14 @@ func (db *DB) Run(b *Batch) error {
 //
 // TODO(pmattis): Allow transaction options to be specified.
 func (db *DB) Tx(retryable func(tx *Tx) error) error {
+	_, file, line, ok := runtime.Caller(1)
```
Comment (tamird) [inline client/db.go]: How expensive is this call?

**Comment by tamird** [comment]
Does this deserve a simple test case?

**Comment by petermattis** [comment]
@tamird Added a test case.

File: client/db.go (hunk 1 above)
Comment (petermattis) [inline client/db.go]: I'll benchmark. Pretty sure it is orders of magnitude less expensive than an RPC.

**Comment by spencerkimball** [comment]
LGTM

**Comment by tamird** [comment]
With a begrudging sigh! LGTM.

File: client/db.go (hunk 1 above)
Comment (petermattis) [inline client/db.go]: `runtime.Caller(1)` takes 500ns on my machine. For comparison, constructing the debug name from the file and line number using `fmt.Sprintf` takes an additional 500ns. So the total debug name construction takes 1us.

Comment (tamird) [inline client/db.go]: Nice, thanks for checking that. The equivalent call was reasonably cheap in MRI but very expensive in JRuby.

**Pull request merged**